docs/glyph_segmentation.md

```diff
@@ -0,0 +1,445 @@
+  segment analysis which in some cases leaves sizable fallback glyph sets. How to implement multi
+  segment analysis is an open question and more development is needed.
+  
+One of the main down sides to this approach is it's reliance on a subsetting closure function which
+are computationally costly. Complex fonts which can require hundreds of closure operation which as a
+result can be slow to process. So another area of open research is if a non closure based approach
+could be developed that is computationally cheaper (for example by producing a segmentation by
+working directly with the substitution and dependencies encoded in a font).
+
+## Goals
+
+The segmentation procedure described in this document aims to achieve the following goals:
+
+* Given a desired segmentation of the font expressed in terms of Unicode code points find a
+  corresponding grouping of glyphs to be made into a set of patches, where each patch adds the data
+  for that set of glyphs.
+
+* Determine the activation conditions for each of those patches. An activation condition is a boolean
+  expression using conjunction and disjunction with the presence of Unicode code points being the boolean
+  values.
+
+* Optimize for minimal data transfer by avoiding duplicating glyphs across patches where possible.
+
+* The chosen glyph segmentation and activation conditions must satisfy the closure requirement:
+
+  The set of glyphs contained in patches loaded for a font subset definition (a set of Unicode
+  code points and a set of layout feature tags) through the patch map tables must be a superset of
+  those in the glyph closure of the font subset definition.
+  
+## Subsetter Glyph Closures
+
+Font subsetters run into a similar problem faced here and solve it with an operation called the glyph
+closure. This takes a subset definition (set of code points and set of feature tags) and calculates
+all glyphs that code be reached by any text using any combination of the input code points.
+
+Closure is a significantly complex process so to save ourselves significant effort we can re-use
+existing subsetter closure implementations to help form segmentations in this process.
+
+For the purposes of this document we define a function:
+
+```
+closure(font, subset_definition) -> {glyph id set}
+```
+
+Where subset_definition is a set of unicode code points and layout feature tags. Subset definitions may
+also alternately be referred to as input segments.
+
+## Segment Closure Analysis
+
+This section defines a procedure to analyze an input segment and determine how it interacts with
+glyphs in the font. It is an extension of the chunking algorithm in Skef’s [binned IFT
+prototype](https://github.com/adobe/binned-ift-reference). Notably it incorporates new capabilities
+of the current IFT spec such as the ability to have conjunction and disjunction in activation
+conditions for patches.
+
+Given:
+
+* An input font, $F$.
+* A list of segments $s_0, …, s_n$. Where:
```

Comment:
In "Goals" above it says the inputs are desired segmentations of Unicode codepoints. I don't see why a desired segmentation would put the same *codepoint* in two segments, so I don't see the purpose of $s_i \neq s_0$ Why not simply specify that the segments must be disjoint? 

Or, if the idea is that the previous stage has done some optimization work, such that it is addressing certain desired duplication cases, why are the input segments in terms of codepoints rather than glyphs (or both)? What if the the earlier stage already has a good idea of where to put unmapped glyphs?

Reply:
s_0 is a special segment. It's the set of codepoints and associated glyphs placed the initial font. The s_i != s_0 is just saying this procedure doesn't apply to the init segment (s_0) since it's considered to be already available and doesn't need patches or conditions generated for it.